app/javascript/mastodon/features/explore/components/story.jsx

```diff
@@ -49,7 +56,8 @@ export default class Story extends PureComponent {
           {thumbnail ? (
             <>
               <div className={classNames('story__thumbnail__preview', { 'story__thumbnail__preview--hidden': thumbnailLoaded })}><Blurhash hash={blurhash} /></div>
-              <img src={thumbnail} onLoad={this.handleImageLoad} alt='' role='presentation' />
+              <img src={thumbnail} onLoad={this.handleImageLoad} alt={thumbnailDescription} title={thumbnailDescription} lang={lang} />
+              {badges}
```

Comment:
Thinking about it a bit more, is there really value to having the ALT badge on these? The alt text itself adds accessibility, but unlike with images in posts, people on Mastodon don't have control over the alt text.

Reply:
To be honest, I am ambivalent about ALT badges. All images on the web ought to have alt texts, but you don't see ALT badges anywhere, because users just assume they exist (I guess).

On social media a lot of images don't have alt texts, because authors are unaware of them. So the badges serves the purpose of (A) telling readers _“this image actually has an alt text unlike most images around here, so it is worth hovering your mouse over this image for a while to reveal the text”_. They also has the side-effect of (B) making authors aware of alt texts so they become more likely to add them to their own posts.

Purpose A is mostly relevant for visually impaired users who might need some help figuring out what exactly is in the picture. Screen readers will see the alt text even without a badge.

I guess purpose A is also relevant for news stories. Purpose B perhaps not so much, because those are a different context.

It is a trade-off. I think the ALT text is relevant information to some users, but it adds UI clutter and partially obscures the image. It is unusual compared to the rest of the web. We could add a user setting, but that also seems like the easy way out (aka give up and let the user bother).

Anyway, I don't have strong opinions on the matter. I added the badges for consistency with other images in the stream, but if you prefer, I am fine with omitting them :-)

Reply:
That's what I meant, purpose B doesn't work here because people linking don't have control over the alt text. I'd say leave out the badges for now and then we can merge it.